### PR TITLE
create cancelupload action

### PR DESCRIPTION
**Pull request by ridwanraji**
[Create upload action](https://github.com/371team4/371Team4/pull/25/files) needs to be merged first.

## Description

- Create cancelUpload state and action
- Create required mutation types
## Related Issue or Trello Card

[Trello card](https://trello.com/c/KoJZRLbt/8-create-service-for-uploading-files-to-firebase-cloud-storage)

## Motivation and Context

- To be able cancel an ongoing file-upload to the firebase storage bucket
## How Has This Been Tested?

- [x] Unit test
- [ ] NightWatch test
- [ ] Coverage testing
- [x] Manual testing

## Test Cases and test paths

- For automatic unit testing, i created a test action/ assertion using Chai assertion library.
- For manual testing, Paired with (Li) to create a cancelUpload button on UploadView component and manually upload files from computer and cancel the upload while its still uploading.
## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by panderson-va** [comment]
There should be screenshots

File: src/store/modules/upload/index.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      context.commit(SET_CANCEL_UPLOAD, false)
+    }, function () {
+      // if upload success
+      alert('upload success')
+      context.commit(SET_IS_UPLOADING, false)
+      context.commit(SET_BYTES_UPLOADED, 0)
+      context.commit(SET_BYTES_REMAINING, 0)
+      context.commit(SET_CANCEL_UPLOAD, false)
+      // store it to event so load event directly use this url not use download api
+      // console.log(upload.snapshot.downloadURL)
```
Comment (panderson-va) [inline src/store/modules/upload/index.js]: Why is this still here?

File: test/unit/specs/Upload.spec.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  // Check if no mutation should have been dispatched
+  if (expectedMutations.length === 0) {
+    expect(count).to.equal(0)
+    done()
+  }
+}
+
+// Unit test for cancelUpload
+describe('index', () => {
```
Comment (panderson-va) [inline test/unit/specs/Upload.spec.js]: Is it possible for a cancel upload to ever fail?

File: src/store/modules/upload/index.js (hunk 1 above)
Comment (ridwanraji) [reply src/store/modules/upload/index.js]: Because i needed an upload going on to be able to cancel upload

Comment (panderson-va) [reply src/store/modules/upload/index.js]: Oh, sorry this comment was only for the commented log line

**Comment by peggyopal** [comment]
@MoranLi He actually shouldn't have any front-end work in this branch, so he shouldn't have the view files

File: src/store/modules/upload/index.js (hunk 1 above)
Comment (peggyopal) [reply src/store/modules/upload/index.js]: I'm still not sure why this log in commented out...

Comment (ridwanraji) [reply src/store/modules/upload/index.js]: I've removed it

**Pull request merged**